src/test/java/gumtree/spoon/TreeTest.java

```diff
@@ -601,4 +619,60 @@ public void test_nestedReturnTypeOfMethodShouldGetParsed() throws Exception {
 		assertEquals(1, returnType.getDescendants().size());
 		assertEquals("java.lang.Integer", returnType.getChild(0).getLabel());
 	}
+
+	private void assertPosition(SourcePosition position, int line, int start, int end) {
+		assertNotNull(position);
+		assertEquals(line, position.getLine());
+		assertEquals(start, position.getSourceStart());
+		assertEquals(end, position.getSourceEnd());
+	}
+
+	@Test
+	public void test_AccessModifierShouldHaveSourcePosition() {
```

Comment:
This method is doing 4 things. In my opinion, it can be 4 different tests.
- CtWrapper in a code string
- CtWrapper in a file
- VirtualElement in a code string
- VirtualElement in a file